### PR TITLE
Youtube Vote Feature, Telegram Api Lib replacement

### PR DESCRIPTION
**Pull request by wtsiamruk**

### CODE REVIEW

**Comment by wtsiamruk** [comment]
Hi @madhead could you please take a look :)

File: src/main/kotlin/by/jprof/telegram/opinions/dao/YoutubeDAO.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    suspend fun removeFromWhiteList(channelId: String) {
+        dynamoDB.deleteItem {
+            it.tableName(whiteListTable)
+            it.key(channelId.toYoutubeWhiteListItem())
+        }.await()
+    }
+
+}
+
+private fun String.toAttributeValue(): AttributeValue = AttributeValue.builder().s(this).build()
```
Comment (madhead) [inline src/main/kotlin/by/jprof/telegram/opinions/dao/YoutubeDAO.kt:37]: Could other DAOs benefit from moving this into a DAO-agnostic extension?

File: src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            val response = youTube.videos().list("snippet,statistics").setId(videoId).execute()
+
+            val videoDetails = response.items.first()
+            val snippet = videoDetails.snippet
+            val channelId = snippet.channelId
+            val title = snippet.title
+            val likes = videoDetails.statistics.likeCount
+            val dislikes = videoDetails.statistics.dislikeCount
+            val rawDescription = if (snippet.description.length > ACCEPTED_DISPLAY_LEN) {
+                snippet.description.substring(IntRange(0, ACCEPTED_DISPLAY_LEN)) + "..."
```
Comment (madhead) [inline src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt:86]:
```suggestion
                snippet.description.substring(IntRange(0, ACCEPTED_DISPLAY_LEN)) + "…"
```

File: src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                    bot.sendMessage(
+                            chatId = update.data.chat.id,
+                            text = videoText,
+                            parseMode = MarkdownV2ParseMode,
+                            replyToMessageId = update.data.messageId,
+                            replyMarkup = InlineKeyboardMarkup(keyboard = votingKeyBoard(votes, videoId))
+                    )
+                } else {
+                    logger.debug("$channelId is not in a white list")
+                    bot.sendMessage(
```
Comment (madhead) [inline src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt:111]: Let's just silently skip those videos?

File: src/test/kotlin/by/jprof/telegram/opinions/processors/YouTubeRegexTest.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package by.jprof.telegram.opinions.processors
+
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+class YouTubeRegexTest {
+     companion object {
+         val siteRegex = """
+            http(?:s?):\/\/(?:www\.)?youtu(?:be\.com\/watch\?v=|\.be\/)([\w\-\_]*)(&(amp;)?‌​[\w\?‌​=]*)?
```
Comment (madhead) [inline src/test/kotlin/by/jprof/telegram/opinions/processors/YouTubeRegexTest.kt:10]: So, are you testing this regexp, or the one from `YoutubeLinksProcessor` ;)

File: src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt (hunk 3 above)
Comment (wtsiamruk) [reply src/main/kotlin/by/jprof/telegram/opinions/processors/YoutubeLinksProcessor.kt:111]: yup, ok :)

File: src/main/kotlin/by/jprof/telegram/opinions/config/youtube.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import org.koin.dsl.module
+
+
+val youtubeModule = module {
+
+    single {
+        val httpTransport = GoogleNetHttpTransport.newTrustedTransport()
+        val jsonFactory = JacksonFactory.getDefaultInstance()
+        val theKey = get<String>(named(YOUTUBE_API_TOKEN))
+        val initializer = YouTubeRequestInitializer(theKey)
```
[the remark this answers is not in the document]
Comment (wtsiamruk) [reply src/main/kotlin/by/jprof/telegram/opinions/config/youtube.kt:17]: IMO it looks more readable that way, even if a little verbose :)

File: src/main/kotlin/by/jprof/telegram/opinions/dao/YoutubeDAO.kt (hunk 1 above)
Comment (wtsiamruk) [reply src/main/kotlin/by/jprof/telegram/opinions/dao/YoutubeDAO.kt:37]: yeah, they might 👍

File: src/intTest/kotlin/by/telegram/opinions/dao/YoutubeDaoIT.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import by.jprof.telegram.opinions.dao.YoutubeDAO
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+
+@ExtendWith(DynamoDB::class)
+internal class YoutubeDaoIT {
```
[the remark this answers is not in the document]
Comment (wtsiamruk) [reply src/intTest/kotlin/by/telegram/opinions/dao/YoutubeDaoIT.kt]: just curiosity

File: build.gradle.kts [hunk 7]
```diff
@@ -12,29 +12,46 @@ repositories {
     maven("https://dl.bintray.com/madhead/maven")
```
[the remark this answers is not in the document]
Comment (wtsiamruk) [reply build.gradle.kts]: done

**Pull request merged**